### DISCUSSION TITLE
Adding form to GraphQL schema in order to submit

### DISCUSSION THREAD

**Discussion by alexjcollins** [solspace/craft-freeform#217] [Questions & Answers] [answered]
Is there any way around this requirement?

We have clients who want to add forms to their site but currently can't set them live until we pull the database from `prod`, apply the schema permissions locally, and then deploy back to `prod`.

**Comment by gustavs-gutmanis** [comment, marked as answer]
Could you please elaborate a bit more on what the issue is?  
What steps are you making and what is the desired outcome and what's happening instead currently.  
I'm sure we can make this work as you expect.

**Reply by alexjcollins** [reply]
Sure!

On a few of our sites, we allow clients to create forms and attach them to entries. The most common scenario is careers pages where a client will create an entry (usually in a channel) for the job role and a matching form in Freeform. The form in Freeform is then linked to the entry using the Freeform Form field type.

At the moment, if we published that entry live, the form would show on the frontend via the relation, but if a user tries to submit, it fails. In order to get the form submitting successfully, it needs to be added to the GraphQL schema, which we need to do locally as that's synced with Project Config.

![image](https://user-images.githubusercontent.com/12031196/137724849-26bb0303-0adc-498c-9ee4-d4d7474322cd.jpeg)

**Reply by gustavs-gutmanis** [reply]
Ok, so if a form is created in the admin panel, it should automatically be added to the GQL schema.
What if you have several GQL schemas? Should it get added to all of them (assuming you have it enabled at all)?
It might be an undesirable effect for some use-cases, so this should probably be a setting you can toggle in freeform settings.

**Reply by alexjcollins** [reply]
Ideally—yes, but you're right, you may not want that on all schemas.
So, the ability to say "Automatically add all forms to X schema" would be really handy.

**Reply by gustavs-gutmanis** [reply]
Alright, we'll look into it, thanks for reporting this @alexjcollins

**Reply by gustavs-gutmanis** [reply]
@alexjcollins this seems to be contrary to what Craft is trying to achieve with these schemas, so unfortunately we won't be able to implement that functionality.

However, if you wish all forms to be available for querying in GQL, just unselect all the custom form checkboxes and leave the `View All Forms` checkbox as the only checked one. This will essentially let you always query every form via GQL, no matter when it was added to the stack.

**Reply by alexjcollins** [reply]
@gustavs-gutmanis Thanks for letting me know about the `View All Forms` permission and how it works
